alpha/lib/model/entryPeer.php

```diff
@@ -796,29 +799,46 @@ public static function retrieveByPKs($pks, PropelPDO $con = null)
 		return entryPeer::doSelect($criteria, $con);
 	}
 
-	public static function filterEntriesByPartnerOrKalturaNetwork(array $entryIds, $partnerId)
+	public static function addValidatedEntry($entryId)
 	{
-		$entryIds = array_slice($entryIds, 0, baseObjectFilter::getMaxInValues());
-		
-		$c = KalturaCriteria::create(entryPeer::OM_CLASS);
-		$c->addAnd(entryPeer::ID, $entryIds, Criteria::IN);
-		
-		if($partnerId >= 0)
-		{
-			$criterionPartnerOrKn = $c->getNewCriterion(entryPeer::PARTNER_ID, $partnerId);
-			$criterionPartnerOrKn->addOr($c->getNewCriterion(entryPeer::DISPLAY_IN_SEARCH, mySearchUtils::DISPLAY_IN_SEARCH_KALTURA_NETWORK));
-			$c->addAnd($criterionPartnerOrKn);
+		$securityContext = array(kCurrentContext::$partner_id, kCurrentContext::$ks);
+		if (self::$lastInitializedContext && self::$lastInitializedContext !== $securityContext) {
+			KalturaLog::log('Reinitalized security context');
+			self::$validatedEntries = array();
 		}
+		
+		self::$validatedEntries[] = $entryId;
+	}
 
-		$dbEntries = self::doSelect($c);
-
-		$entryIds = array();
-		foreach ($dbEntries as $dbEntry)
+	public static function filterEntriesByPartnerOrKalturaNetwork(array $entryIds, $partnerId)
+	{
+		KalturaLog::debug("Valid enties: " . implode(',', self::$validatedEntries));
+		$validatedEntries = array_intersect($entryIds, self::$validatedEntries);
+		$entryIds = array_diff($entryIds, self::$validatedEntries);
+		KalturaLog::debug("More enties to validate: " . implode(',', $entryIds));
+		if(count($entryIds))
 		{
-			$entryIds[] = $dbEntry->getId();
+			$entryIds = array_slice($entryIds, 0, baseObjectFilter::getMaxInValues());
+			
+			$c = KalturaCriteria::create(entryPeer::OM_CLASS);
+			$c->addAnd(entryPeer::ID, $entryIds, Criteria::IN);
+			
+			if($partnerId >= 0)
+			{
+				$criterionPartnerOrKn = $c->getNewCriterion(entryPeer::PARTNER_ID, $partnerId);
+				$criterionPartnerOrKn->addOr($c->getNewCriterion(entryPeer::DISPLAY_IN_SEARCH, mySearchUtils::DISPLAY_IN_SEARCH_KALTURA_NETWORK));
+				$c->addAnd($criterionPartnerOrKn);
+			}
+	
+			$dbEntries = self::doSelect($c);
+	
+			foreach ($dbEntries as $dbEntry)
+			{
+				$validatedEntries[] = $dbEntry->getId();
```

Comment:
It's already limited in line 821: $entryIds = array_slice($entryIds, 0, baseObjectFilter::getMaxInValues());